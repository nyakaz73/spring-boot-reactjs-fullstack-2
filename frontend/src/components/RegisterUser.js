import React, {Component} from 'react';

class RegisterUser extends Component{
    render() {
        return(
            <form>
                <div style={{display:'flex' }}>
                    <input
                        type = "text"
                        name = "name"
                        placeholder="Name"
                        style={leftInput}
                    />

                    <input
                        type = "text"
                        name = "surname"
                        placeholder="Surname"
                        style={rightInput}
                    />
                </div>

                <br/>

                <div style={{display:'flex', }}>
                    <input
                        type = "text"
                        name = "username"
                        placeholder="Username"
                        style={leftInput}
                    />

                    <input
                        type = "text"
                        name = "password"
                        placeholder="Password"
                        style={rightInput}
                    />
                </div>
                <br/>
                <div style={{display:'flex' }}>
                    <input
                        type = "text"
                        name = "email"
                        placeholder="Email"
                        style={leftInput}
                    />
                   <span style={rightInput}></span>
                </div>
                <br/>
                <input
                    type="submit"
                    value="Submit"
                    className="btn"
                />
                <br/>
            </form>
        )
    }
}

const leftInput = {
    flex:'5',
    padding:'5px',
    margin:'10px 10px 0px 0px'
}

const rightInput = {
    flex:'5',
    padding:'5px',
    margin:'10px 0px 0px 10px'
}

export default RegisterUser;