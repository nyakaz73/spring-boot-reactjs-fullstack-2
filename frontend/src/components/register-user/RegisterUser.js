import React, {Component} from 'react';
import PropTypes from "prop-types";
class RegisterUser extends Component{
    constructor(props) {
        super(props);
        this.state = {
            name:'',
            surname:'',
            email:'',
            username:'',
            password:''
        }
        //If you dont use arrow function you will have to manually bind like this
        //If you dont bind you wont be able to access items in the state of this component because it wont be recognised in lifecycle
        //this.onChange = this.onChange.bind(this);
        this.onSubmit = this.onSubmit.bind(this);
    }

    onChange =(e) =>{
        //This is what you do for individual target
        //this.setState({name:e.target.value});
        //But if you have plenty
        this.setState({[e.target.name]:e.target.value})
    }

    onSubmit(e){
        e.preventDefault();
        //Copying state object to newUser
        let newUser = this.state;
        this.props.addUser(newUser);
        //Resetting the fields
        this.setState({
            name:'',
            surname:'',
            email:'',
            username:'',
            password:''
        });
    }
    render() {
        return(
            <form onSubmit={this.onSubmit}>
                <div style={{display:'flex' }}>
                    <input
                        type = "text"
                        name = "name"
                        placeholder="Name"
                        style={leftInput}
                        value={this.state.name}
                        onChange={this.onChange}
                    />

                    <input
                        type = "text"
                        name = "surname"
                        placeholder="Surname"
                        style={rightInput}
                        value={this.state.surname}
                        onChange={this.onChange}
                    />
                </div>
                <br/>
                <div style={{display:'flex', }}>
                    <input
                        type = "text"
                        name = "username"
                        placeholder="Username"
                        style={leftInput}
                        value={this.state.username}
                        onChange={this.onChange}
                    />

                    <input
                        type = "text"
                        name = "password"
                        placeholder="Password"
                        style={rightInput}
                        value={this.state.password}
                        onChange={this.onChange}
                    />
                </div>
                <br/>
                <div style={{display:'flex' }}>
                    <input
                        type = "text"
                        name = "email"
                        placeholder="Email"
                        style={leftInput}
                        value={this.state.email}
                        onChange={this.onChange}
                    />
                   <span style={rightInput}></span>
                </div>
                <br/>
                <input
                    type="submit"
                    value="Submit"
                    className="btn"
                />
                <br/>
            </form>
        )
    }
}

const leftInput = {
    flex:'5',
    padding:'5px',
    margin:'10px 10px 0px 0px'
}

const rightInput = {
    flex:'5',
    padding:'5px',
    margin:'10px 0px 0px 10px'
}

RegisterUser.propTyoes = {
    addUser:PropTypes.func.isRequired,
}

export default RegisterUser;