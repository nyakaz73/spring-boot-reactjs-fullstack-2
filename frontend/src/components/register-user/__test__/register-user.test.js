import React from "react";
import ReactDOM from 'react-dom';
import { unmountComponentAtNode} from 'react-dom';
import RegisterUser from "../RegisterUser";
import {act} from "@testing-library/react";

//for Snapshot testing
import  renderer from 'react-test-renderer';
let container;

beforeEach(()=>{
    container = document.createElement('div')
    document.body.appendChild(container);
});

afterEach(()=>{
    //clean up code after exiting
    unmountComponentAtNode(container);
    document.body.removeChild(container);
    container = null;
});

it('renders RegisterUser component with or without addUser function callback prop without crusing',  ()=> {
    const addUser = jest.fn();
    act(()=>{
       ReactDOM.render(<RegisterUser addUser={addUser}/>,container);
    });

});

it("matches snapshot with No Prop", ()=>{
    const tree= renderer.create(<RegisterUser/>).toJSON();
    expect(tree).toMatchSnapshot();

})

it("matches snapshot with Exactly one addUser callback prop", ()=>{
    const addUser = jest.fn();
    const tree = renderer.create(<RegisterUser addUser={addUser}/>).toJSON();
    expect(tree).toMatchSnapshot();

});