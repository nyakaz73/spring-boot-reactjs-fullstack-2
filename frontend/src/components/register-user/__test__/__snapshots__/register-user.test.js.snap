// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot with Exactly one addUser callback prop 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    style={
      Object {
        "display": "flex",
      }
    }
  >
    <input
      name="name"
      onChange={[Function]}
      placeholder="Name"
      style={
        Object {
          "flex": "5",
          "margin": "10px 10px 0px 0px",
          "padding": "5px",
        }
      }
      type="text"
      value=""
    />
    <input
      name="surname"
      onChange={[Function]}
      placeholder="Surname"
      style={
        Object {
          "flex": "5",
          "margin": "10px 0px 0px 10px",
          "padding": "5px",
        }
      }
      type="text"
      value=""
    />
  </div>
  <br />
  <div
    style={
      Object {
        "display": "flex",
      }
    }
  >
    <input
      name="username"
      onChange={[Function]}
      placeholder="Username"
      style={
        Object {
          "flex": "5",
          "margin": "10px 10px 0px 0px",
          "padding": "5px",
        }
      }
      type="text"
      value=""
    />
    <input
      name="password"
      onChange={[Function]}
      placeholder="Password"
      style={
        Object {
          "flex": "5",
          "margin": "10px 0px 0px 10px",
          "padding": "5px",
        }
      }
      type="text"
      value=""
    />
  </div>
  <br />
  <div
    style={
      Object {
        "display": "flex",
      }
    }
  >
    <input
      name="email"
      onChange={[Function]}
      placeholder="Email"
      style={
        Object {
          "flex": "5",
          "margin": "10px 10px 0px 0px",
          "padding": "5px",
        }
      }
      type="text"
      value=""
    />
    <span
      style={
        Object {
          "flex": "5",
          "margin": "10px 0px 0px 10px",
          "padding": "5px",
        }
      }
    />
  </div>
  <br />
  <input
    className="btn"
    type="submit"
    value="Submit"
  />
  <br />
</form>
`;

exports[`matches snapshot with No Prop 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    style={
      Object {
        "display": "flex",
      }
    }
  >
    <input
      name="name"
      onChange={[Function]}
      placeholder="Name"
      style={
        Object {
          "flex": "5",
          "margin": "10px 10px 0px 0px",
          "padding": "5px",
        }
      }
      type="text"
      value=""
    />
    <input
      name="surname"
      onChange={[Function]}
      placeholder="Surname"
      style={
        Object {
          "flex": "5",
          "margin": "10px 0px 0px 10px",
          "padding": "5px",
        }
      }
      type="text"
      value=""
    />
  </div>
  <br />
  <div
    style={
      Object {
        "display": "flex",
      }
    }
  >
    <input
      name="username"
      onChange={[Function]}
      placeholder="Username"
      style={
        Object {
          "flex": "5",
          "margin": "10px 10px 0px 0px",
          "padding": "5px",
        }
      }
      type="text"
      value=""
    />
    <input
      name="password"
      onChange={[Function]}
      placeholder="Password"
      style={
        Object {
          "flex": "5",
          "margin": "10px 0px 0px 10px",
          "padding": "5px",
        }
      }
      type="text"
      value=""
    />
  </div>
  <br />
  <div
    style={
      Object {
        "display": "flex",
      }
    }
  >
    <input
      name="email"
      onChange={[Function]}
      placeholder="Email"
      style={
        Object {
          "flex": "5",
          "margin": "10px 10px 0px 0px",
          "padding": "5px",
        }
      }
      type="text"
      value=""
    />
    <span
      style={
        Object {
          "flex": "5",
          "margin": "10px 0px 0px 10px",
          "padding": "5px",
        }
      }
    />
  </div>
  <br />
  <input
    className="btn"
    type="submit"
    value="Submit"
  />
  <br />
</form>
`;
